particle-iot/redis
# Optimize prefix matching

Optimize PSUBSCRIBE and PUBLISH for cases where the pattern is a prefix (single wildcard at the end) by using a radix tree to store the patterns. This makes the performance of those commands O(log(N)) instead of O(N).

Steps to validate:
- Run `make` in the root of the directory
- Run `src/redis-server`
- In separate terminals, run several instances of `redis-cli`. Run `psubscribe test*` in some, `psubscribe test-pub*` in others and `publish test-publish 123` in a last. The publish should be received by all the subscribers.
- Close some of the subscribers and publish again. The publish should be received by the remaining subscribers.
